packages/forms/frontend/sirius-components-forms/src/propertysections/PropertySectionLabel.tsx

```diff
@@ -41,21 +47,31 @@ const usePropertySectionLabelStyles = makeStyles((theme) => ({
   },
 }));
 
-export const PropertySectionLabel = ({ editingContextId, formId, widget }: PropertySectionLabelProps) => {
+export const PropertySectionLabel = memo(({ editingContextId, formId, widget }: PropertySectionLabelProps) => {
   const classes = usePropertySectionLabelStyles();
 
+  let propertiesSectionLabelComponent: React.ReactChild = (
+    <div className={classes.propertySectionLabel}>
+      <Typography className={classes.typography} variant="subtitle2">
+        {widget.label}
+      </Typography>
+      {widget.hasHelpText ? (
+        <HelpTooltip editingContextId={editingContextId} formId={formId} widgetId={widget.id} />
+      ) : null}
+    </div>
+  );
+
+  const { data: widgetLabelFormConfigurer } = useData<Array<WidgeLabelConfigurer>>(
+    widgetLabelFormCustomizerExtensionPoint
+  );
```

Comment:
Why do the API look this way? All the extension point do not have to follow this pattern. The last time, we had to transform some data structure, it's not the case now.

Here you have two possible options, on one side you can replace the content of the label or you can add decorators next to the label.

If you want to replace the label with another component, which seems to be what you are doing here, you would need to use `useComponent` and instead use it where the PropertySectionLabel is used. As for my use case, I do not care if a widget do not have any label or help text, I still want to add information to the label anyway. With this code, it would not be possible (see line 72 to 74).

For me the solution could either look like this (in every widget), no need to change the behavior of the PropertySectionLabel:

```
  const { Component: PropertySectionLabel } = useComponent(widgetLabelExtensionPoint);
  return (
    <div>
      <PropertySectionLabel editingContextId={editingContextId} formId={formId} widget={widget} />
      
      CODE OF THE WIDGET
    </div>
  );
```

- Advantages: it's simple, very easy to setup and use
- Drawbacks: only one contribution can replace the existing label

Or we could try another approach by letting others contribute label decorators in PropertySectionLabel. You won't be able to do anything you want but it will let you do some stuff:

```
export const PropertySectionLabel = ({ editingContextId, formId, widget }: PropertySectionLabelProps) => {
  const classes = usePropertySectionLabelStyles();

  const propertySectionPanelDecorators: ComponentExtension<PropertySectionLabelDecoratorProps>[] = useComponents(propertySectionLabelDecoratorExtensionPoint);

  if (!widget.label && !widget.hasHelpText) {
    return (
      <div>
        {propertySectionPanelDecorators.map(({ Component: PropertySectionPanelDecorator }), index) => (
            < PropertySectionPanelDecorator editingContextId={editingContextId} formId={formId} widget={widget} key={index}/>
        )}
      </div>
    );
  } else {
    return (
      <div className={classes.propertySectionLabel}>
        <Typography className={classes.typography} variant="subtitle2">
          {widget.label}
        </Typography>
        {widget.hasHelpText ? (
          <HelpTooltip editingContextId={editingContextId} formId={formId} widgetId={widget.id} />
        ) : null}
        {propertySectionPanelDecorators.map(({ Component: PropertySectionPanelDecorator }), index) => (
            < PropertySectionPanelDecorator editingContextId={editingContextId} formId={formId} widget={widget} key={index}/>
        )}
      </div>
    );
  }
};
```

- Advantages: You can have multiple decorators contributed
- Drawbacks: You cannot change anything else than the decorators but it's not needed right know so who cares

Reply:
Thanks for the feedback, I pushed a version using your second proposal.